inst/app/modules/main_map.R

```diff
@@ -31,20 +82,64 @@ filter_collision_data <- reactive({
 
   data_filtered = filter(data_filtered, Severity %in% input$severity_filter)
 
+  data_filtered = filter(data_filtered, include_pex %in% input$pedestrian_involved_filter)
+
+  # Get the serial numbers (in vector form) where vehicles involved includes users' selected vehicle class
+  serial_no_with_selected_vehicle_class = hk_vehicles %>%
+    filter(Vehicle_Class %in% input$vehicle_class_filter) %>%
+    # convert single column data frame to vector
+    pull(Serial_No_) %>%
+    # remove duplicated serial number if there are more than 1 vehicle class
+    unique()
+
+  data_filtered = filter(data_filtered, Serial_No_ %in% serial_no_with_selected_vehicle_class)
```

Comment:
You are making several `filter()` calls on the same `data_filtered` object. Would it be better to do these multiple calls in one call to simplify the script, or is this not possible?

For example:

```suggestion
  data_filtered = filter(data_filtered, 
                                     include_pex %in% input$pedestrian_involved_filter,
                                     Serial_No_ %in% serial_no_with_selected_vehicle_class)

  # Get the serial numbers (in vector form) where vehicles involved includes users' selected vehicle class
  serial_no_with_selected_vehicle_class = hk_vehicles %>%
    filter(Vehicle_Class %in% input$vehicle_class_filter) %>%
    # convert single column data frame to vector
    pull(Serial_No_) %>%
    # remove duplicated serial number if there are more than 1 vehicle class
    unique()
```

Reply:
Thanks for the suggestions. As each filter may go wrong (parameters in function/UI/dataset etc.), I now separate filter steps one by one to track the errors and for easier debugging. Say, I could add breakpoints to all `data_filtered = filter(...)` lines to check which filter widgets are not functioning correctly.

Reply:
Ahhh, very good point, that makes a lot of sense, happy with this then!